src/Symfony/Component/HttpKernel/EventListener/LocaleListener.php

```diff
@@ -65,6 +67,11 @@ private function setLocale(Request $request)
         if ($locale = $request->attributes->get('_locale')) {
             $request->setLocale($locale);
         }
+
+        if (null === $locale && null !== $this->availableLocales &&
```

Comment:
IIUC, it’s not possible to set ˋavailable_locales` to ˋnull` in the FrameworkBundle configuration, so it will not work. The config option must be nullable.

Reply:
I did the other way around. instead of checking null, I check it's emptiness instead.